backend/controllers/caseNote.ts

```diff
@@ -4,46 +4,53 @@ import { properCaseName } from '../utils'
 
 const getOffenderUrl = (offenderNo) => `/prisoner/${offenderNo}`
 
-export const caseNoteFactory = (prisonApi, caseNotesApi) => {
-  const displayCreateCaseNotePage = async (req, res) => {
-    const { offenderNo } = req.params
-    const { type, subType } = req.query || {}
-    try {
-      const { firstName, lastName } = await prisonApi.getDetails(res.locals, offenderNo)
+const getOffenderDetails = async (prisonApi, res, offenderNo) => {
+  const { firstName, lastName } = await prisonApi.getDetails(res.locals, offenderNo)
 
-      const caseNoteTypes = await caseNotesApi.myCaseNoteTypes(res.locals)
+  return {
+    offenderNo,
+    profileUrl: getOffenderUrl(offenderNo),
+    name: `${properCaseName(firstName)} ${properCaseName(lastName)}`,
+  }
+}
 
-      const types = caseNoteTypes
-        .filter((caseNoteType) => caseNoteType.activeFlag === 'Y')
-        .map((caseNoteType) => ({
-          value: caseNoteType.code,
-          text: caseNoteType.description,
+const getCaseNoteTypes = async (caseNotesApi, res, type) => {
+  const caseNoteTypes = (await caseNotesApi.myCaseNoteTypes(res.locals)).filter(
+    (caseNoteType) => caseNoteType.activeFlag === 'Y'
+  )
+
+  const types = caseNoteTypes.map((caseNoteType) => ({
+    value: caseNoteType.code,
+    text: caseNoteType.description,
+  }))
+
+  const subTypes = caseNoteTypes
+    .filter((caseNoteType) => caseNoteType.code === type)
```

Comment:
only retrieve sub types that match the selected `type` or if `type` is `undefined` then the sub types will be empty